internal/experimental/registry/client.go

```diff
@@ -144,40 +147,54 @@ func (c *Client) PushChart(ref *Reference) error {
 }
 
 // PullChart downloads a chart from a registry
-func (c *Client) PullChart(ref *Reference) error {
+func (c *Client) PullChart(ref *Reference) (*bytes.Buffer, error) {
+	buf := bytes.NewBuffer(nil)
+
 	if ref.Tag == "" {
-		return errors.New("tag explicitly required")
-	}
-	existing, err := c.cache.FetchReference(ref)
-	if err != nil {
-		return err
+		return buf, errors.New("tag explicitly required")
 	}
+
 	fmt.Fprintf(c.out, "%s: Pulling from %s\n", ref.Tag, ref.Repo)
-	manifest, _, err := oras.Pull(ctx(c.out, c.debug), c.resolver, ref.FullName(), c.cache.Ingester(),
+
+	store := content.NewMemoryStore()
```

Comment:
since we want to keep the existing `helm chart pull` functionality, lets create a `PullChartToCache` that contains all the old functionality, then use that in the chart pull action

Reply:
ok if code is duplicated. we will eventually drop `PullChartToCache`